source/client/service_impl.cc

```diff
@@ -0,0 +1,104 @@
+#include "client/service_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include "client/client.h"
+#include "client/options_impl.h"
+
+#include "api/client/options.pb.validate.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void ServiceImpl::nighthawkRunner(const nighthawk::client::ExecutionRequest& request) {
+  Envoy::Thread::LockGuard lock(mutex_);
+  OptionsPtr options = std::make_unique<OptionsImpl>(request.options());
+  Envoy::Thread::MutexBasicLockable log_lock;
+  auto logging_context = std::make_unique<Envoy::Logger::Context>(
+      spdlog::level::from_str(options->verbosity()), "[%T.%f][%t][%L] %v", log_lock);
+  process_ = std::make_unique<ProcessImpl>(*options, time_system_);
```

Comment:
Why do you need to lock process if it's local?

Reply:
Sorry, cleaned this up. The locking was a remnant which is no longer needed.